core/src/main/scala/org/scalarelational/result/QueryResultsIterator.scala

```diff
@@ -44,22 +44,23 @@ class QueryResultsIterator[E, R](rs: ResultSet, val query: Query[E, R]) extends
 
   def next() = nextOption().getOrElse(throw new RuntimeException("No more results. Use nextOption() instead."))
 
-  protected def columnValue[T](rs: ResultSet, index: Int, c: ColumnLike[_], converter: DataType[T]): T = {
-    val value = rs.getObject(index + 1)
+  protected def columnValue[T, S](rs: ResultSet, index: Int, c: ColumnLike[T, S], dataType: DataType[T, S]): T = {
+    val value = rs.getObject(index + 1).asInstanceOf[S]
     if ((c.has(Polymorphic) && !c.isOptional) && value == null) null.asInstanceOf[T]
-    else converter.fromSQLType(c, value)
+    else dataType.converter.fromSQL(c, value)
   }
 
-  protected def valueFromExpressions[T](expression: SelectExpression[T], index: Int): ExpressionValue[T] =
+  protected def valueFromExpressions[T, S](expression: SelectExpression[T], index: Int): ExpressionValue[T] =
     expression match {
-      case column: ColumnLike[T] => {
-        val value = columnValue(rs, index, column, column.dataType)
-        ColumnValue(column, value, None)
+      case column: ColumnLike[_, _] => {
```

Comment:
I can't because the compiler then gives a warning because the generics are erasured.